### ISSUE TITLE
1080Ti Compatibility

### ISSUE THREAD

**Issue by gordan-bobic** [nicehash/excavator#140]
If this could become available soon, it would be much appreciated (1.3.9a on Linux):

$ ./excavator -c 1080Ti-neoscrypt.json
=========================== www.nicehash.com =========================
Excavator v1.3.9a_nvidia GPU Miner for NiceHash.
Copyright (C) 2017 NiceHash. All rights reserved.
Developed by
djeZo, zawawa, dropky, voidstar, and agiz
with help and contributions from
pallas, Vorksholk, bitbandi, ocminer, and Genoil.
=========================== www.nicehash.com =========================

Build time: 2018-01-15 15:45:52+00:00
Build number: 94357606
[20:38:53][0x00007f540c416740][info] Log started
[20:38:53][0x00007f540c416740][error] core | No compatible devices found!

**Comment by dropky** [comment]
1080 Ti is already supported. Do the previous versions have the same problem or is it only with the version 1.3.9a? Can you post here your config file (1080Ti-neoscrypt.json) and nvidia-smi output?

**Comment by gordan-bobic** [comment]
Config file:
```
[
	{"time":0, "commands":[
		{"id":1,"method":"algorithm.add","params":["neoscrypt","neoscrypt.eu.nicehash.com:3341","3Nn2BE41Savb2iW9vJaZmcxewcHVwG9ZyA.1080Ti"]}
	]},
	{"time":1, "commands":[
		{"id":1,"method":"worker.add","params":["0","0"]}
	]},
	{"time":10,"loop":15,"commands":[
		{"id":1,"method":"worker.add","params":["0"]},
		{"id":1,"method":"algorithm.print.speeds","params":["0"]}
	]}
]
```

nvidia-smi:
```
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 387.34                 Driver Version: 387.34                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 108...  On   | 00000000:05:00.0  On |                  N/A |
| 80%   57C    P0   139W / 150W |   3132MiB / 11172MiB |    100%      Default |
+-------------------------------+----------------------+----------------------+
|   1  GeForce GTX 108...  On   | 00000000:83:00.0 Off |                  N/A |
| 85%   73C    P2   149W / 150W |   2561MiB / 11172MiB |    100%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0      7777      C   ./ethminer                                  2535MiB |
|    0     17930      G   /usr/bin/X                                   351MiB |
|    0     26808      G   ...-token=FBC41751F79F2DBDE129F0DB3F7F8563   224MiB |
|    1      2289      G   Xorg                                           5MiB |
|    1     10518      C   ./ethminer                                  2535MiB |
+-----------------------------------------------------------------------------+
```

**Comment by m4rkw** [comment]
1.4.1a build also doesn’t work, these cards are EVGA

**Comment by dropky** [comment]
@gordan-bobic I noticed that you're running ethminer on your cards. Can you remove it and then try to run excavator? Device could be set into exclusive mode, that way once a GPU has a process, no other process can attach to the same GPU.

**Comment by m4rkw** [comment]
Hi, I'm a friend of his, I'm fairly sure he's tried running it without ethminer running

**Comment by gordan-bobic** [comment]
@dropky It doesn't worth even when it is the only thing running on the card, not even Xorg.
It works fine under the exact same circumstances, on the same machine, with the same software stack with lesser cards (1070Ti, 1070, 980Ti).

**Comment by UselessGuru** [comment]
@gordan-bobic: I can see you are running an old driver version.

I have a GTX1080ti in my box and Excavator (all versions up to 1.4.1a) runs flawlessly. I use Nvidia drivers version 390.65 (Windows 10)

**Comment by dropky** [comment]
387 driver should be fine, @gordan-bobic which Linux distribution do you have?

Also, please take a look at the "Note for Linux users" -
 https://github.com/nicehash/excavator/releases/tag/v1.3.7a.

**Comment by m4rkw** [comment]
lol you've been told @gordan-bobic!  ubuntu or GTFO 🤣

**Comment by gordan-bobic** [comment]
@UselessGuru I guess you missed the part where I wasn't using Windows (1st line of the original post).

@dropky It is Ubuntu 16.04

**Comment by dropky** [comment]
@m4rkw lol :D FYI v1.4.1 should work on most linux distributions

**Comment by m4rkw** [comment]
@dropky since other 1080ti cards work it seems like it's possibly an issue with the vendor id or some other kind of id that excavator uses to detect compatible cards?

**Comment by gordan-bobic** [comment]
@dropky  Ubuntu dynamically linked binaries are problematic on most non-ubuntu userspaces, which is why I asked for CentOS builds to be includes. But that can be overcome easily enough using docker containers (or just booting Ubuntu for simplicity).

**Comment by dropky** [comment]
@gordan-bobic We tested v1.4.1 on CentOS 7 and there were no issues. You have to extract the package and find the path to excavator binary :) . We'll take under consideration to add centos builds. 

@m4rkw Excavator apparently doesn't recognize any CUDA devices. Why not? Well... This is interesting since we're using [NVAPI ](https://developer.nvidia.com/nvapi) and it should recognize the same devices as nvidia-smi.

**Comment by gordan-bobic** [comment]
Fixed!

It looks like a file from one of the cuda packages must have been broken. I just took a list of all of all the *nvidia* and *cuda* packages and the files they contained, uninstalled the packages, deleted any leftover files, reinstalled all the same packages (still with 387 drivers) and now excavator finds the 1080Ti.

Bizzare in the extreme, but there you go.
Apologies for the noise. Very strange that _everything_ else worked, and only excavator and only on the 1080Ti were affected. :-/
